### PR TITLE
Release 2.21.0-rc.0

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,34 @@
+## 2.21.0-rc.0 / 2020-08-26
+
+* [CHANGE] Promtool: Add tsdb commands as promtool tsdb subcommand. #6088
+* [FEATURE] Docker Swarm SD: Support tasks and service without published ports. #7686
+* [FEATURE] Eureka SD: New service discovery. #3369
+* [FEATURE] Hetzner SD: New service discovery. #7822
+* [FEATURE] Kubernetes SD: Support Kubernetes EndpointSlices. #6838
+* [ENHANCEMENT] Support composite durations in PromQL, config and UI, e.g. 1h30m. #7713 #7833
+* [ENHANCEMENT] DNS SD: add srv record target and port meta labels. #7678
```
Comment (juliusv) [inline CHANGELOG.md:9]:
```suggestion
* [ENHANCEMENT] DNS SD: Add SRV record target and port meta labels. #7678
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,34 @@
...
+* [ENHANCEMENT] Rules: Label rule_group_iterations metric with group name. #7823
+* [ENHANCEMENT] Scrape: Add per scrape-config targets limit. #7554
+* [ENHANCEMENT] Scrape: Do not stop scrapes in progress during reload. #7752
+* [ENHANCEMENT] UI: Change dropdown hover color to white. #5593
+* [ENHANCEMENT] UI: Move "remove graph" closer to query area. #5990
+* [BUGFIX] Disable HTTP2 because of concerns with the go HTTP2 client. #7588 #7701
+* [BUGFIX] PromQL: Drop metric name in bool comparison between two instant vectors. #7819
+* [BUGFIX] PromQL: query_log_file path is now relative to the config file. #7701
+* [BUGFIX] Rules: Detect extra fields in rule files. #7767
+* [BUGFIX] Rules: Fix bad metrics name. #7787
```
Comment (juliusv) [inline CHANGELOG.md:21]: Maybe change this to be clearer about what it does:

```suggestion
* [BUGFIX] Rules: Disallow overwriting the metric name in the `labels` section of recording rules. #7787
```

**Comment by codesome** [comment]
Good idea to start prombench right after opening the PR instead of after merging it!

**Comment by roidelapluie** [comment]
what can cause the diff of head chunks?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,34 @@
...
+* [ENHANCEMENT] PromQL: Reduce the amount of data queried by remote read when a subquery has an offset. #7667
+* [ENHANCEMENT] Promtool: add --time option to query instant command. #7829
+* [ENHANCEMENT] Rules: Label rule_group_iterations metric with group name. #7823
+* [ENHANCEMENT] Scrape: Add per scrape-config targets limit. #7554
+* [ENHANCEMENT] Scrape: Do not stop scrapes in progress during reload. #7752
+* [ENHANCEMENT] UI: Change dropdown hover color to white. #5593
+* [ENHANCEMENT] UI: Move "remove graph" closer to query area. #5990
+* [BUGFIX] Disable HTTP2 because of concerns with the go HTTP2 client. #7588 #7701
+* [BUGFIX] PromQL: Drop metric name in bool comparison between two instant vectors. #7819
+* [BUGFIX] PromQL: query_log_file path is now relative to the config file. #7701
```
Comment (brian-brazil) [inline CHANGELOG.md:19]: This is also a CHANGE, so should be listed as such up top.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,34 @@
...
+* [CHANGE] Promtool: Add tsdb commands as promtool tsdb subcommand. #6088
+* [FEATURE] Docker Swarm SD: Support tasks and service without published ports. #7686
+* [FEATURE] Eureka SD: New service discovery. #3369
+* [FEATURE] Hetzner SD: New service discovery. #7822
+* [FEATURE] Kubernetes SD: Support Kubernetes EndpointSlices. #6838
+* [ENHANCEMENT] Support composite durations in PromQL, config and UI, e.g. 1h30m. #7713 #7833
+* [ENHANCEMENT] DNS SD: Add SRV record target and port meta labels. #7678
+* [ENHANCEMENT] PromQL: Reduce the amount of data queried by remote read when a subquery has an offset. #7667
+* [ENHANCEMENT] Promtool: add --time option to query instant command. #7829
+* [ENHANCEMENT] Rules: Label rule_group_iterations metric with group name. #7823
```
Comment (brian-brazil) [inline CHANGELOG.md:12]: This is a change, all label changes to non-info metrics are breaking.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,34 @@
...
+* [ENHANCEMENT] Scrape: Add per scrape-config targets limit. #7554
```
Comment (brian-brazil) [inline CHANGELOG.md:13]: Feature

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,34 @@
+## 2.21.0-rc.0 / 2020-08-26
+
+* [CHANGE] Promtool: Add tsdb commands as promtool tsdb subcommand. #6088
+* [FEATURE] Docker Swarm SD: Support tasks and service without published ports. #7686
```
Comment (brian-brazil) [inline CHANGELOG.md:4]: Enhancement, it's a small addition to an existing discovery.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,34 @@
...
+* [FEATURE] Eureka SD: New service discovery. #3369
+* [FEATURE] Hetzner SD: New service discovery. #7822
+* [FEATURE] Kubernetes SD: Support Kubernetes EndpointSlices. #6838
```
Comment (brian-brazil) [inline CHANGELOG.md:7]: Is this big enough to be a feature?

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,34 @@
...
+* [ENHANCEMENT] Support composite durations in PromQL, config and UI, e.g. 1h30m. #7713 #7833
+* [ENHANCEMENT] DNS SD: Add SRV record target and port meta labels. #7678
+* [ENHANCEMENT] PromQL: Reduce the amount of data queried by remote read when a subquery has an offset. #7667
+* [ENHANCEMENT] Promtool: add --time option to query instant command. #7829
+* [ENHANCEMENT] Rules: Label rule_group_iterations metric with group name. #7823
+* [ENHANCEMENT] Scrape: Add per scrape-config targets limit. #7554
+* [ENHANCEMENT] Scrape: Do not stop scrapes in progress during reload. #7752
```
Comment (brian-brazil) [inline CHANGELOG.md:14]: Bugfix, as this was already meant to be the behaviour.

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,34 @@
...
+* [BUGFIX] PromQL: query_log_file path is now relative to the config file. #7701
+* [BUGFIX] Rules: Detect extra fields in rule files. #7767
+* [BUGFIX] Rules: Disallow overwriting the metric name in the `labels` section of recording rules. #7787
+* [BUGFIX] Rules: Keep evaluation timestamp across reloads. #7775
+* [BUGFIX] TSDB: Fix 'chunks.HeadReadWriter: maxt of the files are not set' error. #7856
+* [BUGFIX] Triton SD: Fix a panic when triton_sd_config is nil. #7671 #7701
+* [BUGFIX] UI: Fix react UI bug with series going on and off. #7804
+* [BUGFIX] Web: Stop CMUX and GRPC servers even with stale connections, preventing the server to stop on SIGTERM. #7810
+
+Note: This release is built with go1.15 which deprecates [X.509
```
Comment (brian-brazil) [inline CHANGELOG.md:28]: This sort of note belongs up top, as it could break people.

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,34 @@
...
+* [FEATURE] Hetzner SD: New service discovery. #7822
+* [FEATURE] Kubernetes SD: Support Kubernetes EndpointSlices. #6838
+* [ENHANCEMENT] Support composite durations in PromQL, config and UI, e.g. 1h30m. #7713 #7833
+* [ENHANCEMENT] DNS SD: Add SRV record target and port meta labels. #7678
+* [ENHANCEMENT] PromQL: Reduce the amount of data queried by remote read when a subquery has an offset. #7667
+* [ENHANCEMENT] Promtool: add --time option to query instant command. #7829
+* [ENHANCEMENT] Rules: Label rule_group_iterations metric with group name. #7823
+* [ENHANCEMENT] Scrape: Add per scrape-config targets limit. #7554
+* [ENHANCEMENT] Scrape: Do not stop scrapes in progress during reload. #7752
+* [ENHANCEMENT] UI: Change dropdown hover color to white. #5593
```
Comment (brian-brazil) [inline CHANGELOG.md:15]: Minor UI tweaks like these don't need to be mentioned.

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,34 @@
+## 2.21.0-rc.0 / 2020-08-26
+
+* [CHANGE] Promtool: Add tsdb commands as promtool tsdb subcommand. #6088
```
Comment (brian-brazil) [inline CHANGELOG.md:3]: As worded this is an enhancement, not a breaking change.

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,34 @@
...
+* [ENHANCEMENT] Support composite durations in PromQL, config and UI, e.g. 1h30m. #7713 #7833
+* [ENHANCEMENT] DNS SD: Add SRV record target and port meta labels. #7678
+* [ENHANCEMENT] PromQL: Reduce the amount of data queried by remote read when a subquery has an offset. #7667
+* [ENHANCEMENT] Promtool: add --time option to query instant command. #7829
+* [ENHANCEMENT] Rules: Label rule_group_iterations metric with group name. #7823
+* [ENHANCEMENT] Scrape: Add per scrape-config targets limit. #7554
+* [ENHANCEMENT] Scrape: Do not stop scrapes in progress during reload. #7752
+* [ENHANCEMENT] UI: Change dropdown hover color to white. #5593
+* [ENHANCEMENT] UI: Move "remove graph" closer to query area. #5990
+* [BUGFIX] Disable HTTP2 because of concerns with the go HTTP2 client. #7588 #7701
```
Comment (brian-brazil) [inline CHANGELOG.md:17]: Hmm, this is probably a change

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,3 +1,34 @@
...
+* [BUGFIX] PromQL: Drop metric name in bool comparison between two instant vectors. #7819
+* [BUGFIX] PromQL: query_log_file path is now relative to the config file. #7701
+* [BUGFIX] Rules: Detect extra fields in rule files. #7767
+* [BUGFIX] Rules: Disallow overwriting the metric name in the `labels` section of recording rules. #7787
+* [BUGFIX] Rules: Keep evaluation timestamp across reloads. #7775
+* [BUGFIX] TSDB: Fix 'chunks.HeadReadWriter: maxt of the files are not set' error. #7856
+* [BUGFIX] Triton SD: Fix a panic when triton_sd_config is nil. #7671 #7701
```
Comment (brian-brazil) [inline CHANGELOG.md:24]: I'm not sure why 7701 is listed here.

**Review by brian-brazil — COMMENTED** [review]
7772 is a bugfix

File: CHANGELOG.md (hunk 8 above)
Comment (roidelapluie) [reply CHANGELOG.md:14]: Yes but it was never implemented or documented, so I'm wondering

Comment (brian-brazil) [reply CHANGELOG.md:14]: This isn't the sort of detail that'd be in the user docs, we have told users that reloads are perfectly safe though.

File: CHANGELOG.md (hunk 13 above)
Comment (roidelapluie) [reply CHANGELOG.md:24]: It removes the need for #7671, it would have fixed the bug too

File: CHANGELOG.md (hunk 7 above)
Comment (roidelapluie) [reply CHANGELOG.md:7]: I think so, and I have checked that with @brancz too.

File: CHANGELOG.md (hunk 13 above)
Comment (brian-brazil) [reply CHANGELOG.md:28]: That's not really relevant here, 7671 was the first change to fix this issue.

Comment (roidelapluie) [reply CHANGELOG.md]: ack

**Comment by brian-brazil** [comment]
:+1:

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,3 +1,34 @@
+## 2.21.0-rc.0 / 2020-08-27
+
+This release is built with go1.15 which deprecates [X.509
```
Comment (beorn7) [inline CHANGELOG.md:3]: Nits:
- Comma before the "which".
- We usually write Go versions as "Go 1.15".

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,3 +1,34 @@
...
+CommonName](https://golang.org/doc/go1.15#commonname) in TLS certificates
+validation.
+
+* [CHANGE] Disable HTTP2 because of concerns with the go HTTP2 client. #7588 #7701
```
Comment (beorn7) [inline CHANGELOG.md:7]: Nit: HTTP/2. And I would capitalize "Go".

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,3 +1,34 @@
...
+* [CHANGE] PromQL: query_log_file path is now relative to the config file. #7701
```
Comment (beorn7) [inline CHANGELOG.md:8]: I think it reads more easily if you backtick `query_log_file`.

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,3 +1,34 @@
...
+* [CHANGE] Promtool: Replace the tsdb command line tool by a promtool tsdb subcommand. #6088
+* [CHANGE] Rules: Label rule_group_iterations metric with group name. #7823
```
Comment (beorn7) [inline CHANGELOG.md:10]: Again, I'd backtick `rule_group_iterations`.

File: CHANGELOG.md [hunk 18]
```diff
@@ -1,3 +1,34 @@
...
+* [FEATURE] Eureka SD: New service discovery. #3369
+* [FEATURE] Hetzner SD: New service discovery. #7822
+* [FEATURE] Kubernetes SD: Support Kubernetes EndpointSlices. #6838
+* [FEATURE] Scrape: Add per scrape-config targets limit. #7554
+* [ENHANCEMENT] Support composite durations in PromQL, config and UI, e.g. 1h30m. #7713 #7833
+* [ENHANCEMENT] DNS SD: Add SRV record target and port meta labels. #7678
+* [ENHANCEMENT] Docker Swarm SD: Support tasks and service without published ports. #7686
+* [ENHANCEMENT] PromQL: Reduce the amount of data queried by remote read when a subquery has an offset. #7667
+* [ENHANCEMENT] Promtool: add --time option to query instant command. #7829
```
Comment (beorn7) [inline CHANGELOG.md:19]: Another backtick opportunity to facilitate reading: `--time`

File: CHANGELOG.md [hunk 19]
```diff
@@ -1,3 +1,34 @@
...
+* [ENHANCEMENT] UI: Move "remove graph" closer to query area. #5990
+* [BUGFIX] PromQL: Drop metric name in bool comparison between two instant vectors. #7819
+* [BUGFIX] Rules: Detect extra fields in rule files. #7767
+* [BUGFIX] Rules: Disallow overwriting the metric name in the `labels` section of recording rules. #7787
+* [BUGFIX] Rules: Keep evaluation timestamp across reloads. #7775
+* [BUGFIX] Scrape: Do not stop scrapes in progress during reload. #7752
+* [BUGFIX] TSDB: Fix 'chunks.HeadReadWriter: maxt of the files are not set' error. #7856
```
Comment (beorn7) [inline CHANGELOG.md:26]: I'd replace single quotes with backticks.

**Review by beorn7 — APPROVED** [review]
I have only typographical nits. Fix them as you see fit.

**Pull request merged**